Spelling

jsoref opened the pull request:
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/codeql-action/commit/525c790ab88928c92ff49e731f998468d1ed0e13#commitcomment-47650569

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/codeql-action/commit/42808dbc96d5279ddc197dd9812f29cda3605e6f

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### Merge / deployment checklist

- [ ] Confirm this change is backwards compatible with existing workflows.
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/master/README.md) has been updated if necessary.

jsoref commented on lib/analysis-paths.js:
```diff
@@ -28,7 +28,7 @@ function printPathFiltersWarning(config, logger) {
     if ((config.paths.length !== 0 || config.pathsIgnore.length !== 0) &&
         !config.languages.every(isInterpretedLanguage)) {
-        logger.warning('The "paths"/"paths-ignore" fields of the config only have effect for Javascript and Python');
+        logger.warning('The "paths"/"paths-ignore" fields of the config only have effect for JavaScript and Python');
```
This is a brand

jsoref commented on lib/codeql.test.js:
```diff
@@ -85,7 +85,7 @@ ava_1.default("download codeql bundle cache with different version cached (not p
     });
 });
-ava_1.default('download codeql bundle cache with pinned different version cached if "latests" tools specified', async (t) => {
+ava_1.default('download codeql bundle cache with pinned different version cached if "latest" tools specified', async (t) => {
```
I can understand this being intentional, I don't think it's a valuable spelling divergence.

jsoref commented on lib/init.js:
```diff
@@ -129,7 +129,7 @@ exports.injectWindowsTracer = injectWindowsTracer;
     logger.startGroup("Setup Python dependencies");
     const scriptsFolder = path.resolve(__dirname, "../python-setup");
-    // Setup tools on the Github hosted runners
+    // Setup tools on the GitHub hosted runners
```
This is a brand

jsoref commented on lib/util.js:
```diff
@@ -171,7 +171,7 @@ exports.getCodeQLDatabasePath = getCodeQLDatabasePath;
  * Removes any API prefix or suffix if one is present.
  */
-function parseGithubUrl(inputUrl) {
+function parseGitHubUrl(inputUrl) {
```
This is clearly an API, but if it's an internal API, then it's worth changing.

If it's from an upstream library, I'd be more than happy to try to get it changed.

I can also easily drop the API change from the general brand change -- just let me know.

jsoref commented on src/sarif_v2.1.0_schema.json:
```diff
@@ -587,13 +587,13 @@
 
         "guid": {
-          "description": "A stable, unique identifer for this external properties object, in the form of a GUID.",
+          "description": "A stable, unique identifier for this external properties object, in the form of a GUID.",
```
This is a versioned schema. I can't tell if there's an upstream for this. If there is, let me know and I can try to send the fix to it instead.

Possibly https://github.com/oasis-tcs/sarif-spec ?

aeisenberg replied:
Yes, this typo is also in the original schema itself. 
https://github.com/oasis-tcs/sarif-spec/blob/master/Schemata/sarif-schema-2.1.0.json#L600

It's a non-functional change, but best to leave this file unchanged just in case. If you're feeling motivated, I'm sure a PR in the `sarif-spec` repo would be appreciated.

aeisenberg reviewed (changes requested):
Thanks for submitting your PR! Most of your changes look reasonable.

You need to re-run `npm run build` in order to get all the compiled javascript corrected. Also, see my comment about removing the changes to the sarif spec file.

jsoref commented:
@aeisenberg: yeah, I've already made the PR to them; dropped here. Updated the map.

I also have a set on en_US changes which I imagined might be controversial, so I left them out of this PR.
https://github.com/jsoref/codeql-action/compare/spelling...en-us

This repository appears to have a mix of en_US/en_GB. I generally encourage repositories to use only one flavor of English... I can offer a PR to adopt either flavor.

aeisenberg commented:
Failing build checks are unrelated to this change and are fixed in main.

aeisenberg commented:
> This repository appears to have a mix of en_US/en_GB. I generally encourage repositories to use only one flavor of English... I can offer a PR to adopt either flavor.

I don't think we have a policy on this. Feel free to open a separate PR and we can discuss.

aeisenberg commented:
Rebased and squashed.
